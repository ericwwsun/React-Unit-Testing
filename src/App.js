import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import Homepage from 'views/Homepage';

function App() {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={Homepage} />
        <Route path={`todo/:todoId`} />
      </Switch>
    </Router>
  );
}

export default App;
