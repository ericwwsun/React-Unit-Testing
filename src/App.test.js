import App from './App';

describe('App', () => {
  test('to match snapshot', () => {
    const { asFragment } = renderWrapper(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
