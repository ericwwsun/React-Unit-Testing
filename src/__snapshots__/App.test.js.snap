// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1qpytq4"
  >
    <h1
      class="css-1t3ezx4"
    >
      My TODO List
    </h1>
    <div
      class="css-1pd4htx"
    >
      <input
        class="css-7z6aq6"
        data-testid="input-title"
        name="input-title"
        type="text"
        value=""
      />
      <button
        class="css-nq14pk"
        data-testid="button-create"
      >
        Create
      </button>
    </div>
    <ul
      class="css-1foypjs"
    >
      <hr />
    </ul>
  </div>
</DocumentFragment>
`;
